docs/advanced/atomic-swap.md

```diff
@@ -0,0 +1,38 @@
+# Atomic Swaps
+
+---
+
+## Overview
+
+An Atomic Swap is a smart contract technology which makes possible to exchange coins from two different blockchains without having to trust any third party, for example a centralized exchange. 
+For an atomic swap, a cryptocurrency must have the 'OP_CHECKLOCKTIMEVERIFY' opcode, which makes a timelock possible.
+The other prerequisite is that they need to have the same hash function. That doesn't necessarily mean that they use the same hash functions for mining, but they need to have the same hash function available in their transaction scripting language.
+
+---
+
+## History of Atomic Swaps
+
+Tier Nolan is considered the inventor of atomic swaps. The concept of decentralized exchanges were widely discussed already in 2012. [The first atomic swap](https://blog.decred.org/2017/09/20/On-Chain-Atomic-Swaps/) happened between Decred and Litecoin on September 20, 2017.
```

Comment:
I think these details about Tier Nolan and 2012 should be backed up with some kind of reference material if possible.

I struggle to see why decentralized exchanges are relevant to the history of atomic swaps, unless we provide some explanation of why that is an important detail.

Otherwise I think this "History of Atomic Swaps" section is a bit thin, and could possibly be removed.